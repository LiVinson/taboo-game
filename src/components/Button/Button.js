import React from "react"
import "./Button.scss"


const Button = ({text, handleClick, type}) => {


  return <button className={`button ${type ? "button--"+type: ""}`} onClick={handleClick}>{text}</button>
}
export default Button
