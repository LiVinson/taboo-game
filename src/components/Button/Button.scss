.button {
    border:none;
    color: var(--color-primary);
    background-color: var(--color-off-white);
    font-size:2rem;
    width:47%;
    padding: 1rem 0rem;


    &--lg {

    }

    &--primary-color {
        color: var(--color-off-white);
        background-color: var(--color-primary--light);


        &--lg {

        }
    }


  

}