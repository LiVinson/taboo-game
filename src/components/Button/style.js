import styled from "styled-components"

/*
  3 types of buttons:

  button
  button primary (extension of button)
  button-lg (extension of button) or button-lg primary (tbd)

*/

export const StyledButton = styled.button`
  border: none;
  color: ${(props) => props.theme.color.secondary};
  background-color: ${(props) => props.theme.color.offWhite};
  width: 47%;
  padding: 0.6rem 0rem;
  border-radius: 2px;
  font-size: 2.3rem;
  font-family: var(--font-family--display);
  text-transform: uppercase;

  &:active {
    outline: none;
    filter: brightness(0.8);
    transform: translateY(2px);
  }
`
export const StyledPrimaryButton = styled(StyledButton)`
  color: ${(props) => props.theme.color.offWhite};
  background-color: ${(props) => props.theme.color.secondary};
  width: 47%;
  &:active {
    filter: brightness(1.2);
  }
`
