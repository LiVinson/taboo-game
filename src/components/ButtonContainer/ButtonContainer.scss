.btnContainer {
    margin-top:1.5rem;
    display:flex;
    justify-content: space-between;
    
}