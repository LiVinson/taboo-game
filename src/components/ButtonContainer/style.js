import styled from "styled-components"


export const StyledButtonContainer = styled.div`
  margin-top: 1.5rem;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`