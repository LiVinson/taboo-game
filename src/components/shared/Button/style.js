import styled from 'styled-components'

/*
  4 types of buttons:

  Button (Styled Button)
  Primary Button (StyledPrimaryButton - extension of Button)
  button-lg (extension of button) or button-lg primary (tbd)
  text button

*/

//Off white button used on the left in forms
export const StyledButton = styled.button`
	width: 47%;
	padding: 0.6rem 0rem;
	border: ${(props) => (props.border ? `2px solid ${props.theme.color.primary}` : 'none')};
	border-radius: 2px;
	background-color: ${(props) => props.theme.color.offWhite};
	font-size: 2.3rem;
	font-family: ${(props) => props.theme.text.display};
	text-transform: uppercase;
	color: ${(props) => props.theme.color.primary};	
	
	&:hover {
		transform: translateY(-2px);
	}

	&:active {
		filter: brightness(0.8);
		transform: translateY(2px);
	}

	&:focus {
		border: none;
		outline: solid 2px ${(props) => props.theme.color.accent1};
	}

	&:disabled {	
		filter: brightness(0.6);
		cursor: not-allowed;
	}
`
//Same as Button, with different font color and background and different focus outline color
export const StyledPrimaryButton = styled(StyledButton)`
	color: ${(props) => props.theme.color.offWhite};
	background-color: ${(props) => props.theme.color.accent1};

	&:focus {
		outline: solid 2px ${(props) => props.theme.color.offWhite};
	}	
`


export const StyledLargeButton = styled(StyledButton)`
	width: 100%;
	padding: 1rem 0;
	border: solid 3px ${(props) => props.theme.color.accent1};
	margin-top: ${(props) => props.margin && '1.2rem'};
	background-color: ${(props) => props.theme.color.lightGray2};
	font-size: 3rem;
	font-weight: 600;	
	color: ${(props) => props.theme.color.primary};
	
	&:focus {
		outline: none;
		border: solid 3px ${(props) => props.theme.color.grayDark1};
	}

`

//Not currently used. TBD if needed
export const StyledTextButton = styled.button`
	display: inline-block;
	font-size: 2.5rem;
	font-weight: 400;
	background-color: inherit;
	color: ${(props) => props.theme.color.primary};
	border: none;
	padding: 0.8rem;
	padding-bottom: 0rem;
	font-family: ${(props) => props.theme.text.display};
	width: fit-content;

	&::after {
		content: '';
		display: block;
		height: 2px;
		width: 100%;
		background-color: currentColor;
		margin-top: 0.5rem;
	}

	&:visited {
		color: ${(props) => props.theme.color.primary};
	}

	&:active,
	&:focus {
		outline: none;
		color: ${(props) => props.theme.color.grayDark2};
	}
`
