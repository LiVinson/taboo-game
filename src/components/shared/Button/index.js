import React from "react"
import { StyledButton, StyledPrimaryButton, StyledLargeButton, StyledTextButton} from "./style"

//Takes in object of properties passed down. Typically includes type, onClick, text, and any other required attributes
export const Button = (props) => {
  return <StyledButton {...props}>{props.text}</StyledButton>
}


export const PrimaryButton = (props) => {
  return <StyledPrimaryButton {...props}>{props.text}</StyledPrimaryButton>
}

export const LargeButton = (props) => {
  return <StyledLargeButton {...props}>{props.text}</StyledLargeButton>
}

export const TextButton = (props) => {
  return <StyledTextButton {...props}>{props.text}</StyledTextButton>
}