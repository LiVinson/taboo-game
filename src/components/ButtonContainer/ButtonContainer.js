import React from "react"
import PropTypes from "prop-types"
import { Button, PrimaryButton, LargeButton } from "components/shared/Button"
import { StyledButtonContainer } from "./style"

//Appears at bottom of Taboo Card when a buttons prop is passed to taboo.
// Receives array of objects with button text and click handler
const ButtonContainer = ({ buttons }) => {
  return (
    <StyledButtonContainer>
      {buttons.map((buttonInfo, index, buttons) =>
        determineButtonType(buttonInfo, index, buttons.length)
      )}
    </StyledButtonContainer>
  )
}

ButtonContainer.propTypes = {
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      onClick: PropTypes.func.IsRequired,
      type: PropTypes.string,
    })
  ),
}

/*3 possible buttons: regular, primary, and large regular
  When there are 2 or more buttons, first is always button (white) and second is primary button (color).
  If there are less than 2 buttons or more than 2 (e.g. 1 or 3 buttons) then that button is large button
  Margin is only added to large button when there are multiple buttons to separate it from buttons above
*/
const determineButtonType = (buttonInfo, index, length) => {

  if (index === 0 && length >= 2) {
    return <Button key={index} {...buttonInfo} />
  } else if (index === 1 && length >= 2) {
    return <PrimaryButton key={index} {...buttonInfo} />
  } else {
    return <LargeButton margin={length>2} key={index} {...buttonInfo} />
  }
}

export default ButtonContainer