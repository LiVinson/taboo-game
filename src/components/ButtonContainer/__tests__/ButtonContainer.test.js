import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import ButtonContainer from '../index'
import { Button, PrimaryButton, LargeButton } from 'components/shared/Button'

describe('Button Container renders and functions correctly', () => {
	const defaultProps = {
		buttons: [
			{ text: 'Back', onClick: jest.fn() },
			{ text: 'Next', onClick: jest.fn() },
		],
	}

	test('Button Container renders correctly', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<ButtonContainer {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Renders Button and PrimaryButton when info for 2 buttons provided', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<ButtonContainer {...props} />)

    expect(wrapper.find(Button)).toHaveLength(1)
    expect(wrapper.find(PrimaryButton)).toHaveLength(1)
    expect(wrapper.find(LargeButton)).toHaveLength(0)
		})

	test('Renders Button, PrimaryButton, and LargeBytton when info for 3 buttons provided', () => {
		const props = {
      buttons: [...defaultProps.buttons, 	{ text: 'Play!', onClick: jest.fn() }],
    }

		const wrapper = shallow(
				<ButtonContainer {...props} />
		)

    expect(wrapper.find(Button)).toHaveLength(1)
    expect(wrapper.find(PrimaryButton)).toHaveLength(1)
    expect(wrapper.find(LargeButton)).toHaveLength(1)
	})
})
