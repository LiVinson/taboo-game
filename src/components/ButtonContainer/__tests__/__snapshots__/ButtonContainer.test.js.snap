// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Container renders and functions correctly Button Container renders correctly 1`] = `
.c1 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c1:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c1:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c1:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c1:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c2 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c2:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c2:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c2:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c2:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c2:focus {
  outline: solid 2px #f7f9f9;
}

.c0 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={[MockFunction]}
  >
    Back
  </button>
  <button
    className="c2"
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
